jerry-core/vm/vm.c

```diff
@@ -2010,10 +2001,37 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         {
           if (ecma_are_values_integer_numbers (left_value, right_value))
           {
-            ecma_integer_value_t left_integer = (ecma_integer_value_t) left_value;
-            ecma_integer_value_t right_integer = (ecma_integer_value_t) right_value;
+            bool is_less = (ecma_integer_value_t) left_value < (ecma_integer_value_t) right_value;
+
+            if (*byte_code_p <= CBC_BRANCH_IF_TRUE_BACKWARD_3 && *byte_code_p >= CBC_BRANCH_IF_TRUE_BACKWARD)
```

Comment:
BTW, might be worth adding a note here that this is actually a lookahead to the next opcode and executing a CBC_BRANCH_IF_TRUE_BACKWARD* right here if that's what is found (I'm not sure that such lookaheads are used anywhere else, so it might be good mentioning.)